ansible/roles/bootstrap/tasks/main.yml

```diff
@@ -0,0 +1,44 @@
+- name: Ensures bootkube dir exists
+  file: path=/tmp/bootkube state=directory
+
+- name: Extract bootkube binaries
+  unarchive:
+    src: https://github.com/kubernetes-incubator/bootkube/releases/download/v0.3.9/bootkube.tar.gz
+    dest: /tmp/bootkube
+    remote_src: True
+
+- name: Render bootkube manifests
+  command: /tmp/bootkube/bin/linux/bootkube render --asset-dir=/tmp/bootkube/assets --experimental-self-hosted-etcd --etcd-servers=http://10.3.0.15:2379 --api-servers=https://{{ inventory_hostname }}:443
```

Comment:
`10.3.0.15` is the fixed incluster-address of the etcd-server, currently set by bootkube itself. Similar to the DNS-Server in kubernetes, which operates at a fixed address. Not really necessary to make this configurable IMO.

To your other point, yes exactly. For now the work happening in upstream kubernetes assumes that DNS is present in a datacenter. To make DNS bootstrap itself and then distribute it to clients we would need to find a solution/workaround ourselves. For distribution something like this can be leveraged: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/bootstrap-discovery.md#method-out-of-band